### PR TITLE
Address TODO items in preparation for 3.0 beta (8/many)

### PR DESCRIPTION
**Pull request by koikonom**

### CODE REVIEW

**Review by mbfrahry — APPROVED** [review]
LGTM!

File: internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go [hunk 1]
```diff
@@ -48,7 +48,9 @@ func resourceOrchestratedVirtualMachineScaleSet() *pluginsdk.Resource {
 			Delete: pluginsdk.DefaultTimeout(60 * time.Minute),
 		},
 
-		// TODO: remove support the the legacy Orchestrated Virtual Machine Scale Set in 3.0
+		// The plan was to remove support the the legacy Orchestrated Virtual Machine Scale Set in 3.0.
+		// Turns out it's still in use
+		// TODO: Revisit in 4.0
```
Comment (tombuildsstuff) [inline internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go:53]: @WodansSon what's the plan for this one?

File: internal/services/batch/batch_pool_data_source.go [hunk 2]
```diff
@@ -362,6 +261,112 @@ func dataSourceBatchPool() *pluginsdk.Resource {
...
+	if !features.ThreePointOhBeta() {
+		s["max_task_retry_count"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeInt,
+			Computed: true,
+		}
+
+		s["environment"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeMap,
+			Optional: true,
+			// Computed: true,
```
Comment (tombuildsstuff) [inline internal/services/batch/batch_pool_data_source.go:361]: pedantic but we can probably remove this?

```suggestion
```

**Review by tombuildsstuff — APPROVED** [review]
One comment for @WodansSon but this otherwise LGTM 👍

**Pull request merged**